### PR TITLE
Add code formatter and linter

### PR DESCRIPTION
**Pull request by nelsonmestevao**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2,64 +2,91 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-    <modelVersion>4.0.0</modelVersion>
+    <modelversion>4.0.0</modelversion>
```
Comment (ruioliveira02) [inline pom.xml]: pom.xml is case sensitive and therefore this does not compile. Exclude the pom from the linter :pray:

File: pom.xml [hunk 2]
```diff
@@ -2,64 +2,91 @@
...
+                    <encoding>${project.encoding}</encoding>
+                    <configlocation>.checkstyle.xml</configlocation>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupid>net.revelc.code.formatter</groupid>
+                <artifactid>formatter-maven-plugin</artifactid>
+                <version>2.10.0</version>
+                <configuration>
+                    <encoding>${project.encoding}</encoding>
```
Comment (ruioliveira02) [inline pom.xml]: Add
```
 <excludes>
      <exclude>pom.xml</exclude>
</excludes>
```
to exclude pom.xml ([Source](https://code.revelc.net/formatter-maven-plugin/examples.html))

**Review by ruioliveira02 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (nelsonmestevao) [reply pom.xml]: I must have formatted with another tool. It is fixed now 💪

File: pom.xml (hunk 1 above)
Comment (nelsonmestevao) [reply pom.xml]: Sorry, I think I run prettier or something here. It is fixed now :raised_hands:

**Comment by nelsonmestevao** [comment]
@ruioliveira02 please, test it now :pray:

**Review by ruioliveira02 — APPROVED** [review]
There is just a small discrepancy in the java version in the docker image but I will fix it in a new PR.

**Pull request merged**